### ISSUE TITLE
Unexpected behavior of `@include` directive on a fragment 

### ISSUE THREAD

**Issue by lwasyl** [apollographql/apollo-kotlin#5338]
### Version

3.8.2
4.0.0-beta.1
**fixed in 4.0.0-beta.2**

### Summary

I observe unexpected behavior with how `@include` directive is handled: when I have two fragments on the same type, and I `@include(if: false)` one of them, then the whole query doesn't return from the cache. Quick outline:
```graphql
query Foo(doInclude: Boolean!) {
  foo {
    bar {
      ...firstFragmentOnBar
      ...secondFragmentOnBar
      ...thirdFragmentOnBar@include(if: doInclude)
    }
  }
}

fragment firstFragmentOnBar on Bar {
  fieldforFirstFragment { ... }
}

fragment secondFragmentOnBar on Bar {
  fieldforSecondFragment { ... }
}

fragment thirdFragmentOnBar on Bar {
  fieldforThirdFragment { ... }
}
```

with the above when I observe `Foo(false)` from the cache, and another query refreshes it, then the first cache-only query doesn't emit proper data. I was pretty confused since I did see data in the normalized cache, there was no error or anything. I feel it's a bug because when I move the `@include` inside the fragment, it works as expected:
```graphql
query Foo(doInclude: Boolean!) {
  foo {
    bar {
      ...firstFragmentOnBar
      ...secondFragmentOnBar
      ...thirdFragmentOnBar
    }
  }
}

fragment firstFragmentOnBar on Bar {
  fieldforFirstFragment { ... }
}

fragment secondFragmentOnBar on Bar {
  fieldforSecondFragment { ... }
}

fragment thirdFragmentOnBar on Bar {
  fieldforThirdFragment@include(if: doInclude) { ... }
}
```

### Logs

```
(Your logs here)
```

**Comment by BoD** [comment]
This has been resolved by #5296

**Comment by BoD** [comment]
Closing as the fix is in 4.0.0-beta.2
